File: src/backend/CareFirstClinic.API/CareFirstClinic.API/Models/ServiceOrder.cs
using System.ComponentModel.DataAnnotations;

namespace CareFirstClinic.API.Models
{
    public enum ServiceOrderStatus { Pending, InProgress, Completed, Cancelled }

    public class ServiceOrder
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid AppointmentId { get; set; }
        public Guid ServiceId { get; set; }
        
        public decimal PriceAtOrder { get; set; }
        public ServiceOrderStatus Status { get; set; } = ServiceOrderStatus.Pending;
        
        // Locking mechanism
        public Guid? LockedByDoctorId { get; set; }
        public DateTime? LockedAt { get; set; }
        
        public string? ResultData { get; set; } // JSON results
        
        public Appointment? Appointment { get; set; }
        public Service? Service { get; set; }
        public Doctor? LockedByDoctor { get; set; }
    }
}

File: src/frontend/src/types/serviceOrder.ts
export interface ServiceField {
  id: string;
  fieldName: string;
  unit?: string;
  dataType?: string;
}

export interface Service {
  id: string;
  name: string;
  price: number;
  description?: string;
  specialtyId?: string;
  isActive: boolean;
  fields: ServiceField[];
}

export interface ServiceOrder {
  id: string;
  appointmentId: string;
  serviceId: string;
  serviceName: string;
  priceAtOrder: number;
  status: 'Pending' | 'InProgress' | 'Completed' | 'Cancelled';
  lockedByDoctorId?: string;
  lockedByDoctorName?: string;
  lockedAt?: string;
  resultData?: string; // JSON string containing { [fieldName]: string | number }
  patientName?: string;
  serviceFields?: ServiceField[];
}

export interface UpdateServiceOrderResultDTO {
  resultData: string;
}

export interface CreateServiceDTO {
  name: string;
  price: number;
  description?: string;
  specialtyId?: string;
  fields: { fieldName: string; unit?: string; dataType: string }[];
}
